src/resolver/mod.rs

```diff
@@ -198,8 +209,8 @@ mod tests {
         return vec![some_host_entry(), other_host_entry()];
     }
 
-    #[test]
-    fn resolve_handles_static_addresses() {
+    #[tokio::test]
+    async fn resolve_handles_static_addresses() {
```

Comment:
When `resolve` is synchronous again there the tests can be synchronous again too.